conan/tools/cmake/toolchain/blocks.py

```diff
@@ -840,8 +840,20 @@ def __init__(self, conanfile, toolchain, items=None):
             for name, block in items:
                 self._blocks[name] = block(conanfile, toolchain)
 
-    def remove(self, name):
+    def keys(self):
+        return self._blocks.keys()
+
+    def items(self):
+        return self._blocks.items()
+
+    def remove(self, name, *args):
         del self._blocks[name]
+        for arg in args:
+            del self._blocks[arg]
+
+    def filter(self, name, *args):
+        to_keep = [name] + list(args)
+        self._blocks = OrderedDict((k, v) for k, v in self._blocks.items() if k in to_keep)
```

Comment:
IMO, the name `filter` sounds like this function is going to return something. As far as I see, this filter is going to perform a kind of `include` or `keep` those names that you're passing, isn't it?

```python
def include(self, name, *args):  # Or directly def include(self, *args):
# Or
def keep(self, name, *args):  # Or directly def keep(self, *args):
```

Reply:
Let's do ``select()``